test/zkbob/ZkBobPoolETH.t.sol

```diff
@@ -0,0 +1,580 @@
+        "PermitTransferFrom(TokenPermissions permitted,address spender,uint256 nonce,uint256 deadline)TokenPermissions(address token,uint256 amount)"
+    );
+
+    ZkBobPoolETH pool;
+    ZkBobDirectDepositQueueETH queue;
+    IWETH9 token;
+    IOperatorManager operatorManager;
+    IPermit2 permit2;
+
+    event Message(uint256 indexed index, bytes32 indexed hash, bytes message);
+
+    event SubmitDirectDeposit(
+        address indexed sender,
+        uint256 indexed nonce,
+        address fallbackUser,
+        ZkAddress.ZkAddress zkAddress,
+        uint64 deposit
+    );
+    event RefundDirectDeposit(uint256 indexed nonce, address receiver, uint256 amount);
+    event CompleteDirectDepositBatch(uint256[] indices);
+
+    function setUp() public {
+        vm.createSelectFork(forkRpcUrl, forkBlock);
+        token = IWETH9(weth);
+        permit2 = IPermit2(permit2Address);
+
+        EIP1967Proxy poolProxy = new EIP1967Proxy(address(this), address(0xdead), "");
+        EIP1967Proxy queueProxy = new EIP1967Proxy(address(this), address(0xdead), "");
+
+        console2.log(weth, address(token));
+        ZkBobPoolETH impl =
+        new ZkBobPoolETH(0, address(token), new TransferVerifierMock(), new TreeUpdateVerifierMock(), new BatchDepositVerifierMock(), address(queueProxy), permit2Address);
+
+        bytes memory initData = abi.encodeWithSelector(
+            ZkBobPool.initialize.selector,
+            initialRoot,
+            1_000_000 ether,
+            100_000 ether,
+            100_000 ether,
+            10_000 ether,
+            10_000 ether,
+            0,
+            0
+        );
+        poolProxy.upgradeToAndCall(address(impl), initData);
+        pool = ZkBobPoolETH(payable(address(poolProxy)));
+
+        ZkBobDirectDepositQueueETH queueImpl = new ZkBobDirectDepositQueueETH(address(pool), address(token));
+        queueProxy.upgradeTo(address(queueImpl));
+        queue = ZkBobDirectDepositQueueETH(address(queueProxy));
+
+        operatorManager = new MutableOperatorManager(user2, user3, "https://example.com");
+        pool.setOperatorManager(operatorManager);
+
+        queue.setOperatorManager(operatorManager);
+        queue.setDirectDepositFee(0.1 gwei);
+        queue.setDirectDepositTimeout(1 days);
+
+        deal(weth, user1, 1 ether);
+        vm.startPrank(user1);
+        token.approve(permit2Address, type(uint256).max);
+        token.approve(address(queue), type(uint256).max);
+        vm.stopPrank();
+
+        vm.startPrank(user2);
+        token.approve(permit2Address, type(uint256).max);
+        token.approve(address(queue), type(uint256).max);
+        vm.stopPrank();
+
+        vm.startPrank(user3);
+        token.approve(permit2Address, type(uint256).max);
+        token.approve(address(queue), type(uint256).max);
+        vm.stopPrank();
+    }
+
+    function testSimpleTransaction() public {
+        bytes memory data1 = _encodePermitDeposit(0.5 ether, 0.01 ether);
```

Comment:
Since the order of permit deposits does not mater with permit2 consider to make an explicit test for this. For example, there could be like this:
```
bytes memory data2 = _encodePermitDeposit(0.75 ether, 0.01 ether);
bytes memory data1 = _encodePermitDeposit(0.5 ether, 0.01 ether);

_transact(data1);
_transact(data2);
```